src/pages/ReimbursementAccount/ReimbursementAccountPage.js

```diff
@@ -144,6 +148,14 @@ class ReimbursementAccountPage extends React.Component {
         }
     }
 
+    fetchData() {
+        // We can specify a step to navigate to by using route params when the component mounts.
+        const stepToOpen = this.getStepToOpenFromRouteParams();
```

Comment:
I'm not 100% sure, but I think if the component has just went from offline and back to online that we would not need to pass a "stepToOpen". It's been a while since I've looked at how that works though. Unsure if it will make a difference.

Reply:
I suggest `stepToOpen` needs to be a global variable, and its only assigned in `componentDidMount()`, what do you think?

Reply:
To clarify, your suggestion is to assign the `stepToOpen` in `componentDidMount()` and then use it again in `componentDidUpdate()`? That seems like a solution to something, but I'm not sure what exactly.

Any idea about why we should do that and whether it matters?

Reply:
Not entirely sure, so i guess we can leave it as it is. We can revisit it anytime.

Reply:
Sorry, I don't feel comfortable with that. I'd like to know whether it matters and be able to produce a clear explanation if anyone asks why we made this change.

Reply:
>To clarify, your suggestion is to assign the stepToOpen in componentDidMount() and then use it again in componentDidUpdate()? That seems like a solution to something, but I'm not sure what exactly.
Any idea about why we should do that and whether it matters?

So I think before a page goes offline, it was once  in an online state, which means if we keep `stepToOpen` as a global variable, it would be set long ago before the page went offline, so when the page comes back online we pass that value to `fetchData()`.

The `stepToOpen` variable is crucial if you wanna 'connect manually' to a bank account in terms of the steps

Reply:
Ok so if we don't set it will we be routed back to an incorrect view when connectivity resumes?

Reply:
Nothing happens, but it's the variable am 💯 certain won't change, so it doesn't really matter.

Reply:
Ok so I am not sure if my point is being made, but just going to explain how I see this working 😄 

- `stepToOpen` is parsed from the route params which _should_ keep the user on the current step if we have one
- if we do not do this when we go offline and regain connectivity they would get redirected whatever step is most appropriate based on the state of the bank account (which is based on the state of the account in the server)

So we actually **do need the logic that you have here**. If we only use the `stepToOpen` when the component mounts then when the account refreshes it could redirect us to the wrong step.

Here's an example of where it makes a difference:

- Add manual bank account
- Company step
- Go offline
- Go online
- Get navigated back to incorrect step

Can test this by moving the logic which sets up `stepToOpen` to happen only in `componentDidMount()`